types/validator.go

```diff
@@ -32,6 +32,25 @@ func NewValidator(pubKey crypto.PubKey, votingPower int64) *Validator {
 	}
 }
 
+func (v *Validator) ValidateBasic() error {
+	if v == nil {
+		return errors.New("nil validator")
+	}
+	if v.PubKey == nil {
+		return errors.New("validator does not have a public key")
+	}
+
+	if v.VotingPower < 0 {
```

Comment:
nowhere do we specify that VotingPower can't be negative... although we probably should